File: 2020/01-report-repair/main.py
import time

def solve():
	f = open("input.txt", "r")
	lines = f.readlines()
	lines = [int(line.strip()) for line in lines]

	for i in range(len(lines)):
		for j in range(i+1, len(lines)):
			# Part 1
			if lines[i] + lines[j] == 2020:
				p1 = (lines[i] * lines[j])
			# Part 2
			for k in range(j+1, len(lines)):
				if lines[i] + lines[j] + lines[k] == 2020:
					p2 = (lines[i] * lines[j] * lines[k])
	f.close()
	return p1, p2

print("Day 01: Report Repair")
start = time.time()
p1, p2 = solve()
print("\tPart 1: ", p1)
print("\tPart 2: ", p2)
print("Done in %s seconds" % round(time.time() - start, 3))

File: 2020/01-report-repair/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

func solve(f string) (p1, p2 int) {
	nums := read(f)

	for i, n1 := range nums {
		for j, n2 := range nums[i:] {
			if n1+n2 == 2020 {
				p1 = n1 * n2
			}
			for _, n3 := range nums[j:] {
				if n1+n2+n3 == 2020 {
					p2 = n1 * n2 * n3
				}
			}
		}
	}

	return p1, p2
}

func read(filename string) []int {
	fc, _ := os.ReadFile(filename)

	lines := strings.Split(string(fc), "\n")
	nums := []int{}
	for _, line := range lines {
		num, _ := strconv.Atoi(line)
		nums = append(nums, num)
	}
	return nums
}

func main() {
	start := time.Now()
	fmt.Println("Day 01: Report Repair")
	p1, p2 := solve("input.txt")
	fmt.Println("\tPart One:", p1) // 1006875
	fmt.Println("\tPart Two:", p2) // 165026160
	fmt.Printf("Done in %.3f seconds \n", time.Since(start).Seconds())
}
